config/initializers/paperclip.rb

```diff
@@ -3,20 +3,25 @@
 upload_path = "uploads/:class/:attachment/:id_partition/:style/:filename"
 
 # We always do local storage in testing for speed and so that we don't rely on external services.
-if !Rails.env.test? && Settings.key?(:paperclip) && Settings.paperclip.storage == "cloud"
-  raise "AWS must be provided if storage set to 'cloud'" unless Settings.key?(:aws)
+storage_type = ENV["NEMO_PAPERCLIP_STORAGE_TYPE"]
+storage_type ||= Settings.key?(:paperclip) ? Settings.paperclip.storage : "local"
+
+if !Rails.env.test? && Settings.key?(:paperclip) && storage_type == "cloud"
+  unless Settings.key?(:aws) || ENV.key?("NEMO_AWS_ACCESS_KEY_ID")
+    raise "AWS settings must be provided if storage set to 'cloud'"
+  end
 
   Paperclip::Attachment.default_options.merge!(
     path: upload_path,
     storage: "fog",
     fog_credentials: {
       provider: "AWS",
-      aws_access_key_id: Settings.aws.access_key_id,
-      aws_secret_access_key: Settings.aws.secret_access_key,
-      region: Settings.aws.region,
+      aws_access_key_id: ENV["NEMO_AWS_ACCESS_KEY_ID"] || Settings.aws.access_key_id,
+      aws_secret_access_key: ENV["NEMO_AWS_SECRET_ACCESS_KEY"] || Settings.aws.secret_access_key,
+      region: ENV["NEMO_AWS_REGION"] || Settings.aws.region,
```

Comment:
I alluded to this at standup, but I would love to keep all ENV config as named variables rather than accessing dynamically every time we use them. We could call it `Settings` or `Constants` or anything, that doesn't matter to me. For a few reasons
1. Easier to type `Settings.foo` vs. `ENV["FOO"]`
1. Easier to autocomplete assigned variables vs. plain strings
1. Allows defaulting them in one place and using the default later. For example I could see `ENV["NEMO_OFFLINE_MODE"] == "true"` proliferating, then one day we want to default it to `"true"` in some specific scenario if it's undefined
1. Allows viewing all possible ENV settings in one place. For example if we add a new feature flag `ENV["SPECIAL_FEATURE"]` but forget to add it to `.env.example`, the only way a new developer would know about it would be grep the whole codebase for `"ENV"` and figure out which settings are possible to use. I've run into this far too many times on open source codebases!

`.env` would still be the one single place where servers are configured, but `Settings` would be where Rails learns about the config. Thoughts?